pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -477,23 +464,13 @@ func (c *Controller) Run(stop <-chan struct{}) {
 		c.networksWatcher.AddNetworksHandler(c.initNetworkLookup)
 		c.initNetworkLookup()
 	}
-
-	go func() {
-		cache.WaitForCacheSync(stop, c.HasSynced)
-		c.queue.Run(stop)
-	}()
-
-	go c.services.Run(stop)
+	go c.serviceInformer.Run(stop)
 	go c.pods.informer.Run(stop)
 	go c.nodeMetadataInformer.Run(stop)
 	go c.filteredNodeInformer.Run(stop)
-
-	// To avoid endpoints without labels or ports, wait for sync.
```

Comment:
this seems intentionally before endpoints, any reason its safe to remove now?

Reply:
As we can see, the actual handler will be executed after cache sync in line 483, so it is same now
https://github.com/istio/istio/pull/23745/files#diff-288be3f23828adceb9ebb059ce53a23eL483